src/components/Filters.js

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import Data from "./Data.json";
+import SearchByName from "./SearchByName";
+
+// Ref. https://www.youtube.com/watch?v=pQndRHoHd4Y
+function Filters() {
+  const [pets, setPets] = useState(Data);
+
+  const filteredResults = (petType) => {
+    if (petType !== "") {
+      const filteredPets = pets.filter((pet) => {
+        return pet.type === petType;
+      });
+      setPets(filteredPets);
+      console.log("filteredPets", filteredPets);
+    } else {
+      setPets(Data);
+      console.log("Data", Data);
+    }
+  };
+
+  return (
+    <div className="centerBox">
+      <button id="sort-by-name">
+        <SearchByName />
+      </button>
+      <div className="buttons">
+        <button id="filter-cats" onClick={() => filteredResults("cat")}>
+          Filter Cats
+        </button>
+        <button id="filter-dogs" onClick={() => filteredResults("dog")}>
+          Filter Dogs
+        </button>
+        <button id="filter-turtles" onClick={() => filteredResults("turtle")}>
+          Filter Turtles
+        </button>
+        <button id="filter-rabbits" onClick={() => filteredResults("rabbit")}>
+          Filter Rabbits
+        </button>
+        <button id="reset" onClick={() => filteredResults("")}>
+          Reset
+        </button>
+        {pets.map((pet, index) => (
+          <div key={index} className="card">
```

Comment:
nit: I would use just the pet name as the key (since they seem to be unique), it's probably not a big deal for an application of this size, but using the index of items that may change order is not recommended (https://legacy.reactjs.org/docs/lists-and-keys.html#:~:text=We%20don%E2%80%99t%20recommend%20using%20indexes%20for%20keys%20if%20the%20order%20of%20items%20may%20change.)